File: android/src/main/kotlin/video/pano/rtc/flutter/ResultCallback.kt
package video.pano.rtc.flutter

import com.pano.rtc.api.Constants
import io.flutter.plugin.common.MethodChannel.Result
import video.pano.rtc.native.Callback

class ResultCallback(private val result: Result?) : Callback() {
    
    override fun success(data: Any?) {
        if (data is Constants.QResult) {
            result?.success(data.value)
        } else {
            if (data == null) {
                failure("NullResult", "")
            } else {
                result?.success(data)
            }
        }
    }

    override fun failure(code: String, message: String) {
        result?.error(code, message, null)
    }
}

File: ios/Classes/ResultCallback.swift
//
//  ResultCallback.swift
//  pano_rtc
//
//  Copyright © 2020 Pano. All rights reserved.
//

import Foundation
import PanoRtc

class ResultCallback: NSObject, Callback {
    private var result: FlutterResult?

    init(_ result: FlutterResult?) {
        self.result = result
    }

    func success(_ data: Any?) {
        result?(data)
    }

    func failure(_ code: String) {
        result?(FlutterError.init(code: code, message: nil, details: nil))
    }
}
